cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/TetheringStore.java

```diff
@@ -224,6 +248,58 @@ public PeerInfo getPeer(String peerName) throws IOException, PeerNotFoundExcepti
     }, PeerNotFoundException.class, IOException.class);
   }
 
+  /**
+   * Initialize last message id sent and received from each peer in {@code peers}
+   * @param peers list of peer names
+   * @param messageIdFromPeer map of peer name to last message id received. It is initialized by this method
+   * @param messageIdToPeer map of peer name to last message id sent. It is initialized by this method
+   */
+  public void initializeSubscriberState(List<String> peers, Map<String, String> messageIdFromPeer,
```

Comment:
avoid modifying input when possible, the contract is not as clear and it's possible to introduce bugs in refactoring if the input maps are not modifiable. A cleaner way is to make this return the information instead. Can introduce a new class to represent subscriber state, it would probably make things more readable than handling multiple Maps anyway.

Reply:
Added a new SubscriberState class. Fixed initializeSubscriberState() to return Subscriberstate